### PR TITLE
DEV: Allow plugins to add extra icons to category-link

### PR DESCRIPTION
**Pull request by markvanlan**
Screenshot from 2020-05-11 13-15-01

This allows for plugins to check a property on a category, add additional icons to the category-link html

### CODE REVIEW

File: app/assets/javascripts/discourse/app/models/category.js [hunk 1]
```diff
@@ -510,6 +510,11 @@ Category.reopenClass({
     return _.sortBy(data, category => {
       return category.get("read_restricted");
     });
+  },
+
+  extraBadgeIconRenderers: [],
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/models/category.js:515]: If these are only used in the helper I'd recommend moving them into there, and adding a `plugin-api` method to add one.

Comment (markvanlan) [reply app/assets/javascripts/discourse/app/models/category.js:515]: Oh, I didnt think about adding the actual array to the PluginApi. Okay, great!

File: app/assets/javascripts/discourse/app/helpers/category-link.js [hunk 2]
```diff
@@ -7,6 +7,12 @@ import Site from "discourse/models/site";
 import { escapeExpression } from "discourse/lib/utilities";
 import { htmlSafe } from "@ember/template";
 
+const CategoryLink = EmberObject.extend({});
```
Comment (eviltrout) [inline app/assets/javascripts/discourse/app/helpers/category-link.js:10]: It doesn't need to be a class does it? The pattern we normally use here is to create a local instance variable:

`let _extraIconRenderers = []` 

It'll be scoped to this module.

Then you can export a function to add a new one to that local variable.

**Comment by markvanlan** [comment]
@eviltrout Third time is the charm here I think

File: app/assets/javascripts/discourse/app/helpers/category-link.js [hunk 3]
```diff
@@ -149,12 +155,15 @@ function defaultCategoryLinkRenderer(category, opts) {
...
 
   if (restricted) {
-    html += `${iconHTML(
-      "lock"
-    )}<span class="category-name" ${categoryDir}>${categoryName}</span>`;
-  } else {
-    html += `<span class="category-name" ${categoryDir}>${categoryName}</span>`;
+    html += iconHTML("lock");
   }
+  _extraIconRenderers.forEach(function(renderer) {
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/helpers/category-link.js:160]: should be fat arrow probably

Comment (markvanlan) [reply app/assets/javascripts/discourse/app/helpers/category-link.js:160]: I literally never know.

**Pull request merged**